src/sentry/event_manager.py

```diff
@@ -973,7 +1015,7 @@ def _save_aggregate(event, flat_hashes, release, **kwargs):
             state=GroupHash.State.LOCKED_IN_MIGRATION
         ).update(group=group)
 
-        if group_is_new and len(new_hashes) == len(all_hashes):
+        if group_is_new and len(new_hashes) >= len(flat_grouphashes):
```

Comment:
`new_hashes` can now be larger than `flat_grouphashes` in case `new_hashes` also contained the hierarchical hash. Previously `all_hashes` was a superset of `new_hashes`.

Reply:
I'm not sure if this condition makes sense. Why is this the condition we want?

Reply:
hmm actually

```suggestion
        if group_is_new and len(new_hashes) == len(to_update):
```

does that make more sense to you?